x-pack/plugins/cases/server/services/attachments/index.ts

```diff
@@ -251,12 +250,11 @@ export class AttachmentService {
       );
 
       const soExtractor = getAttachmentSOExtractor(res.attributes);
-      const so = soExtractor.populateFieldsFromReferences<AttachmentAttributes>(res);
-      const injectedAttributes = injectPersistableReferencesToSO(so.attributes, so.references, {
-        persistableStateAttachmentTypeRegistry: this.persistableStateAttachmentTypeRegistry,
-      });
-
-      return { ...so, attributes: { ...so.attributes, ...injectedAttributes } };
+      return injectAttachmentSOReferences(
```

Comment:
Would it make sense to move `getAttachmentSOExtractor` directly into the `inject*` method? I think we can call it from there since we have `res` right?